### ISSUE TITLE
Clean up NumPyro DM mass-selection API before v0.1.0

### ISSUE THREAD

**Issue by gomeshun** [gomeshun/jeanspy#31]
Follow-up to #29. The numerical/autodiff blocker is resolved, but the public NumPyro Jeans API still exposes `use_analytic_dm: Optional[bool] = None`, and the model-aware `auto` policy is encoded via `_default_enclosed_mass_method` rather than an explicit autodiff-safety capability.

Please implement this cleanup before v0.1.0.

## Required API change

Replace `use_analytic_dm` on all three public/internal Jeans solver entry points with a method-style option:

```python
dm_mass_method: str = "auto"
```

Supported values must be exactly:

- `"auto"`
- `"analytic"`
- `"numeric"`

Apply consistently to:

- `DSphModel.sigmalos2`
- `DSphModel.sigmalos2_kernel`
- `DSphModel.sigmalos2_abel`

There were no pre-v0.1 repository callers that require preserving `use_analytic_dm`, so remove it rather than keeping a three-state compatibility shim unless a current-main caller proves otherwise.

## Capability-driven auto selection

Make `DMModel.enclosed_mass(method="auto")` capability-driven rather than relying on a model-specific default-method string. Prefer an explicit semantic capability such as:

```python
analytic_enclosed_mass_autodiff_safe = False
```

on `DMModel`, overridden to `True` for `NFWModel`. General `ZhaoModel` should remain `False` because its analytic `jax.scipy.special.betainc` path is not autodiff-safe with respect to Zhao shape parameters `(a, b, g)`.

Then resolve `method="auto"` as:

- analytic only if an analytic enclosed-mass implementation exists **and** the model declares that analytic path autodiff-safe;
- otherwise numeric.

Keep explicit `method="analytic"` and `method="numeric"` overrides available.

## Validation / behavior

Preserve the intended behavior introduced in #29:

- NFW + auto -> analytic mass;
- general Zhao + auto -> numeric mass;
- explicit Zhao analytic remains available for value/regression work when shape-parameter gradients are not required.

Validate method strings consistently and raise a clear `ValueError` for invalid values.

## Tests

Update/add tests so they cover at least:

1. NFW `method="auto"` selects/matches analytic.
2. Zhao `method="auto"` selects/matches numeric.
3. Explicit Zhao numeric vs analytic value comparison remains.
4. Default `DSphModel.sigmalos2` Zhao path still has a finite `jax.grad` through at least one Zhao shape parameter.
5. `dm_mass_method="analytic"` and `dm_mass_method="numeric"` explicitly propagate through kernel and Abel paths as intended.
6. Invalid `dm_mass_method` is rejected.
7. No active source/tests/docs still use `use_analytic_dm`.

Update cache/JIT static keys if necessary so `dm_mass_method` participates correctly in cached solver behavior.

## Documentation

Update README/docstrings from `use_analytic_dm` to `dm_mass_method`, while preserving the explanation that Zhao's analytic `betainc` expression is not JAX-autodiff-safe in its shape parameters and that `auto` is the NUTS-safe default.

## Final checks

Start from current `main`, keep the scope limited to this API cleanup, run focused tests and the full suite, and report exact results. Also run/search for leftover `use_analytic_dm` references before finalizing.

@copilot Please implement this issue on a branch and open a PR for review.
